src/app/account-details/account-details.resolver.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot, Router } from '@angular/router';
+import { UserService } from '../core/user.service';
+
+
+@Injectable()
+export class AccountDetailsResolver implements Resolve<any> {
+
+  constructor(public userService: UserService, private router: Router) { }
+
+  resolve(route: ActivatedRouteSnapshot): Promise<[any]> {
+    let userDetails: any
+    return new Promise((resolve, reject) => {
+      this.userService.fetchUserData()
```

Comment:
fetchUserData returns promise, no sense wrap it in outer promise. Remove this outer and this resolve and reject inside

Reply:
Not sure how to do it, as this.resolve leads to infinity loop here